DataProcessing/dataHandler.py

```diff
@@ -0,0 +1,125 @@
+import os.path
+import os
+import os.path as osp
+import tempfile
+from argparse import ArgumentParser
+
+import mmcv
+import numpy as np
+from mmtrack.apis import inference_mot, init_model
+from cv2 import imwrite
+import os, glob, shutil
+import tensorflow as tf
+import matplotlib.pyplot as plt
+from sklearn.cluster import KMeans
+from sklearn.metrics import silhouette_score
+import cv2
+
+class DataHandler:
+
+    def __init__(self, 
+                 input_folder_path:str,
+                 output_video_path:str,
+                 config,
+                 checkpoint,
+                 device,
+                 ):
+
+        self.input_path = input_folder_path
+        self.output_path = output_video_path
+        self.config = config
+        self.checkpoint = checkpoint
+        self.device = device
+        self.crop_folder_path = self.output_path + 'crops/'
+        self.cluster_folder_path = self.output_path + 'clusters/'
+        self.cam_id = 2
+
+    def track_persons_and_crop_from_dir(self, capture_index:int=500, acc_threshold:int=0.999) -> None:
+        """
+        capture_index = frame will be captured every capture_index value. input 1 for all frames
+        """
+        def create_crops_folder() -> None:
```

Comment:
not sure we need it as a function, considering that it is a two lines long function which is called only once